engine/collection/pusher/engine.go

```
@@ -53,55 +83,86 @@ func New(log zerolog.Logger, net network.EngineRegistry, state protocol.State, e
 	}
 	e.conduit = conduit
 
+	e.cm = component.NewComponentManagerBuilder().
+		AddWorker(e.inboundMessageWorker).
+		Build()
+	e.Component = e.cm
+
 	return e, nil
 }
 
-// Ready returns a ready channel that is closed once the engine has fully
-// started.
-func (e *Engine) Ready() <-chan struct{} {
-	return e.unit.Ready()
+func (e *Engine) inboundMessageWorker(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+	ready()
+
+	done := ctx.Done()
+	wake := e.notifier.Channel()
+	for {
+		select {
+		case <-done:
+			return
+		case <-wake:
+			e.processInboundMessages(ctx)
+		}
+	}
 }
 
-// Done returns a done channel that is closed once the engine has fully stopped.
-func (e *Engine) Done() <-chan struct{} {
-	return e.unit.Done()
+func (e *Engine) processInboundMessages(ctx context.Context) {
+	for {
+		nextMessage, ok := e.inbound.Pop()
+		if !ok {
+			return
+		}
+
+		asEngineWrapper := nextMessage.(*engine.Message)
+		asSCGMsg := asEngineWrapper.Payload.(*messages.SubmitCollectionGuarantee)
+		originID := asEngineWrapper.OriginID
+
+		_ = e.process(originID, asSCGMsg)
```

Seems incorrect to fully discard any error, should likely be logged instead?